### ISSUE TITLE
freee access for lan and not for others

### ISSUE THREAD

**Issue by boscorelly** [BookStackApp/BookStack#5377]
### Attempted Debugging

- [X] I have read the debugging page

### Searched GitHub Issues

- [X] I have searched GitHub for the issue.

### Describe the Scenario

hi,
i haven't found the answer : is it possible to create a scenario where a specific ip range can access publicly the documentation, while others must login ?

### Exact BookStack Version

last

### Hosting Environment

docker

**Comment by ssddanbrown** [comment]
Hi @boscorelly,
No, that's not something which is supported at all.

It could be possible to hack on an instance in various ways using the [logical theme system](https://github.com/BookStackApp/BookStack/blob/development/dev/docs/logical-theme-system.md), but I'm hesitant to provide further guidance on that as it's easy to result in an insecure setup when it comes to customizations around authentication.

**Comment by boscorelly** [comment]
Hi,The aim is to let users freely access the docs when at the office, and force them to login through office 365 (where passwordless mfa is mandatory) when they are roaming 
    
    Envoyé à partir de Outlook pour iOS
  

 De : Dan Brown ***@***.***>Envoyé : mardi, décembre 31, 2024 4:47 PMÀ : BookStackApp/BookStack ***@***.***>Cc : Camille Ollié ***@***.***>; Mention ***@***.***>Objet : Re: [BookStackApp/BookStack] freee access for lan and not for others (Issue #5377) 
Hi @boscorelly,
No, that's not something which is supported at all.
It could be possible to hack on an instance in various ways using the logical theme system, but I'm hesitant to provide further guidance on that as it's easy to result in an insecure setup when it comes to customizations around authentication.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
Envoyé à partir de Outlook pour iOS

**Comment by ssddanbrown** [comment]
Sure, I understand the aim.
Would users not already be generally logged into Office365 while in the office though? If so, just using Azure/MS based OIDC for SSO may provide a low-friction route for use.

**Comment by boscorelly** [comment]
Yes they are, but they will not be auto logged in.And my users will find that complex to have to log in (yes, we are French)
    
    Envoyé à partir de Outlook pour iOS
  

 De : Dan Brown ***@***.***>Envoyé : mercredi, janvier 1, 2025 4:22 PMÀ : BookStackApp/BookStack ***@***.***>Cc : Camille Ollié ***@***.***>; Mention ***@***.***>Objet : Re: [BookStackApp/BookStack] freee access for lan and not for others (Issue #5377) 
Sure, I understand the aim.
Would users not already be generally logged into Office365 while in the office though? If so, just using Azure/MS based OIDC for SSO may provide a low-friction route for use.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
Envoyé à partir de Outlook pour iOS

**Comment by ssddanbrown** [comment]
> but they will not be auto logged in.

Our OIDC/SAML options do provide the option to auto-start the auth process, which results in automatic login assuming they're already logged into the upstream auth system in that browser.

Otherwise, I'll close this off since the original question has been answered.

**Comment by boscorelly** [comment]
Let me try it with saml, i will tell you. 
    
    Envoyé à partir de Outlook pour iOS
  

 De : Dan Brown ***@***.***>Envoyé : vendredi, janvier 3, 2025 10:01 PMÀ : BookStackApp/BookStack ***@***.***>Cc : Camille Ollié ***@***.***>; Mention ***@***.***>Objet : Re: [BookStackApp/BookStack] freee access for lan and not for others (Issue #5377) 

but they will not be auto logged in.

Our OIDC/SAML options do provide the option to auto-start the auth process, which results in automatic login assuming they're already logged into the upstream auth system in that browser.
Otherwise, I'll close this off since the original question has been answered.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
Envoyé à partir de Outlook pour iOS

**Comment by boscorelly** [comment]
hi,

i tried and it works like i want.
